src/olympia/api/authentication.py

```diff
@@ -142,7 +142,11 @@ def authenticate_credentials(self, payload):
                 'User tried to authenticate with invalid auth hash in'
                 'payload {}'.format(payload)
             )
-            raise exceptions.AuthenticationFailed()
+            msg = {
+                'detail': gettext('Auth hash mismatch. Session is likely expired.'),
```

Comment:
Not a big fan of that messaging, but if it's going to be replaced by something more generic on the frontend maybe that's fine... It's tough to pick something given it's a fairly technical issue...